### ISSUE TITLE
Replace SETTINGS object with proper redux state

### ISSUE THREAD

**Issue by rhysyngsun** [mitodl/open-discussions#945]
We currently store initial data in a global `SETTINGS` object, which is not ideal (globals are bad) and requires us to completely reload the application should any of this data change (e.g. a login/registration completes). 

What we would probably want to be able to achieve with this (could be split into a few PRs):

- [ ] Somehow pass this data into the application from the first pageload (not sure on best practices around this)
- [ ] Be able to reload this data for the current user, ideally through an API call.

Difficult areas are probably going to be anywhere we're using values under `SETTINGS` deeper than top level containers. Might be worth shaping this data differently than we currently do (e.g. have an objection just for feature flags), so we don't have to pass around this entire object all the time.

We probably want to also put some forward-thinking design around this, because we follow this `SETTINGS` pattern in other apps too and we may want to update them likewise.

**Comment by aliceriot** [comment]
I agree we want to move away from this. Maybe if we included the `SETTINGS` object in the initial HTML view we send to the client we could then use that as the initial state for the store when the app boots. There might be a better way to handle that though, not sure.

I think we may want to split it into different reducers for different pieces of state, but we might be able to also use the redux selector pattern profitably here to make it easy to get out the value of, say, a feature flag - then if we needed that value in a component we could connect it to the store doing something like


```js
const MyComponent = () => (
  <JSX! />
)

export connect(state => ({
    enabled: getFeatureFlag[ENABLE_MY_COMPONENT],
  }),
  MyComponent
)
```

Or something along those lines. I think it wouldn't be too bad to connect components (even non-container components) if we did that, and using selectors and similar things to reduce boilerplate would probably also make things like this easier to test.
